src/scaffold/entry_type/integrity.rs

```diff
@@ -829,9 +835,12 @@ fn add_entry_type_to_validation_arms(
                                                                         )
                                                                     {
                                                                         let new_arm: syn::Arm = syn::parse_str(
-                                                                             format!(
-"(EntryTypes::{pascal_entry_def_name}({snake_entry_def_name}), EntryTypes::{pascal_entry_def_name}(original_{snake_entry_def_name})) => 
-    validate_update_{snake_entry_def_name}(action, {snake_entry_def_name}, original_action, original_{snake_entry_def_name}),", 
+                                                                             format!(r#"
+EntryTypes::{pascal_entry_def_name}({snake_entry_def_name}) => {{
+    let original_app_entry = must_get_valid_record(action.clone().original_action_address)?;
+    let original_{snake_entry_def_name} = {pascal_entry_def_name}::try_from(original_app_entry)?;
```

Comment:
> In case it is indeed possible to get different entry types here in validation it might be worth adding a more meaningful error message here as well.

Should a `wasm_error` be emitted or should a `Ok(ValidateCallbackResult::Invalid("message".to_string()))` be returned instead?

Reply:
I'm actually not sure how the implications of the two compare, i.e. I'm not sure how holochain treats a wasm error differently than a Invalid result.

Reply:
Returning a wasm error will make Holochain retry. Returning Ok with an Invalid message will cause the dependency to be marked as invalid.

Reply:
Oh okay...so it probably should be marked as invalid? Or is there any failure mode where that deserialization could fail although the entry type is actually correct?

Reply:
@matthme also, was this error `Original and updated entry types must be the same` as a result of converting a `Record` into an entry type defined in the `EntryTypes` enum?